### PR TITLE
feat(pagination): pagination styles added to toolkit

### PR DESCRIPTION
**Pull request by cathysiller**
JIRA:  https://jira.rax.io/browse/SURF-676

### LGTM's
- [x] Dev LGTM
- [x] Design LGTM
- [x] Zoom LGTM

# Pagination
pagination-helixui

### CODE REVIEW

File: docs/components/pagination/index.html [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+        <button class="hxBtn" disabled>
+          <hx-icon type="angle-start"></hx-icon>
+        </button>
+        <button class="hxBtn" disabled>
+          <hx-icon type="angle-left"></hx-icon>
+        </button>
+        <button class="hxBtn" aria-current="true">1</button>
+        <button class="hxBtn">2</button>
+        <button class="hxBtn">3</button>
+         <button class="hxBtn">4</button>
```
Comment (nimbinatus) [inline docs/components/pagination/index.html]: Check indent; looks off in review mode:
screen shot 2018-03-27 at 11 04 16 am

File: docs/components/pagination/index.html [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+        </button>
+        <button class="hxBtn">
+          <hx-icon type="angle-end"></hx-icon>
+        </button>
+      </div>
+    {% endcode %}
+  </section>
+
+  <section>
+    <h2 class="hxSectionTitle" id="second-page">Second Page</h2>
```
Comment (nimbinatus) [inline docs/components/pagination/index.html]: So this isn't the second page selected; it's the third page. Am I misunderstanding what this shows?

File: docs/components/pagination/index.html [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+        </button>
+        <button class="hxBtn">
+          <hx-icon type="angle-end"></hx-icon>
+        </button>
+      </div>
+    {% endcode %}
+  </section>
+
+  <section>
+    <h2 class="hxSectionTitle" id="middle-page">Middle Page</h2>
```
Comment (nimbinatus) [inline docs/components/pagination/index.html]: Last page?

File: docs/components/pagination/index.html [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+    <h2 class="hxSectionTitle" id="second-page">Second Page</h2>
+    <div class="demo">
+      <div class="hxPagination hxBtnGroup">
+        <button class="hxBtn">
+          <hx-icon type="angle-start"></hx-icon>
+        </button>
+        <button class="hxBtn">
+          <hx-icon type="angle-left"></hx-icon>
+        </button>
+        <button class="hxBtn">8</button>
```
Comment (nimbinatus) [inline docs/components/pagination/index.html]: Are the numbers off here? It starts at 8, but the code sample starts at 1.

File: docs/components/pagination/index.html (hunk 1 above)
Comment (cathysiller) [reply docs/components/pagination/index.html]: good catch -- fixed

File: docs/components/pagination/index.html (hunk 2 above)
Comment (cathysiller) [reply docs/components/pagination/index.html]: good catch -- its supposed to be an example of a selected middle page -- fixed to represent that so order should be first, middle, last

File: docs/components/pagination/index.html (hunk 3 above)
Comment (cathysiller) [reply docs/components/pagination/index.html]: yes good catch!

File: docs/components/pagination/index.html (hunk 4 above)
Comment (cathysiller) [reply docs/components/pagination/index.html]: so for the middle page example we wanted the example to represents single digit and double digit numbers so pagination should start at 8, end at 12, and 10 should be selected

File: docs/components/pagination/index.html [hunk 5]
```diff
@@ -0,0 +1,146 @@
+---
+title: Pagination
+also:
+  components/buttons: Buttons
```
Comment (CITguy) [inline docs/components/pagination/index.html]: Add link to Icons component, too.

File: docs/components/pagination/index.html [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+---
+{% extends 'component.njk' %}
+{% block content %}
+  <section>
+    <h2 class="hxSectionTitle" id="first-page">First Page</h2>
```
Comment (CITguy) [inline docs/components/pagination/index.html]: Remove class from heading.

File: docs/components/pagination/index.html [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+    <div class="demo">
+      <div class="hxPagination hxBtnGroup">
+        <button class="hxBtn" disabled>
+          <hx-icon type="angle-start"></hx-icon>
+        </button>
+        <button class="hxBtn" disabled>
+          <hx-icon type="angle-left"></hx-icon>
+        </button>
+        <button class="hxBtn" aria-current="true">1</button>
```
Comment (CITguy) [inline docs/components/pagination/index.html]: We need a blurb somewhere to note that the `aria-current` attribute needs added to apply style as well as denote semantic meaning. Probably at the top of the page, below the page title (similar to Text Inputs).

File: docs/components/pagination/index.html (hunk 3 above)
Comment (CITguy) [reply docs/components/pagination/index.html]: Please remove class from title.

File: docs/components/pagination/index.html [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+    <h2 class="hxSectionTitle" id="last-page">Last Page</h2>
+    <div class="demo">
+      <div class="hxPagination hxBtnGroup">
+        <button class="hxBtn">
+          <hx-icon type="angle-start"></hx-icon>
+        </button>
+        <button class="hxBtn">
+          <hx-icon type="angle-left"></hx-icon>
+        </button>
+        <button class="hxBtn">1</button>
```
Comment (CITguy) [inline docs/components/pagination/index.html]: Update page numbers to span from 16-20. This makes the progression in the docs go from "the first 5 pages" (1-5) to the "middle 5 pages" (8-12) then the "last 5 pages" (16-20).

File: src/helix-ui/styles/modules/pagination.less [hunk 9]
```diff
@@ -0,0 +1,19 @@
+.hxPagination {
+  background-color: @gray-0;
+  border: 1px solid @gray-300;
+  border-radius: 2px;
+
+  button {
+    border-color: transparent;
+    min-width: 2.50rem; //40px
```
Comment (CITguy) [inline src/helix-ui/styles/modules/pagination.less]: *NOTE*: (@araiford) this is not exactly to spec, but we found that it works better when transitioning between 1 and 2-digit page numbers.

**Comment by araiford** [comment]
This is looking great @cathysiller 

The only nit pick I have is that we've got the hxbtn set to use a transparent border color when inside the pagination bar.  That's making the total height of the bar 36px.  The spec in the sketch doc is 32px.

Can we set `border: none` for .hxPagination .hxBtn?  That would get it down to 34px tall, which doesn't look bad given the inherent widths of the buttons.

**Comment by araiford** [comment]
Thanks for the change @cathysiller 

Design LGTM

**Pull request merged**